pre-piksi-multi-launch-updates.md

```diff
@@ -0,0 +1,121 @@
+### SBP changes before shipping Piksi Multi
+
+There are a few changes we would like to make in the SBP protocol. These changes
+will be much more painful to make after we ship and customers are relying on
+the protocol to be largely static, so we'd like to come to a consensus on
+these and implement them before shipping Piksi Multi.
+
+These changes attempt to accomplish a few things:
+- Make messages more bandwidth efficient
+- Make user side logic of using messages more obvious
+- Remove state from the protocol, where possible
+
+### 1. Replace obs forwarding with general purpose forwarding message
+
+Current there is a convention in the protocol that if the sender_id of a
+MSG_OBS is 0, it means it is a forwarded observation from another device. We
+propose to replace this with a general purpose forwarding message that wraps
+the forwarded message, discarding the preamble and checksum of the forwarded
+message. This way, the sender_id of the forwarded message will be preserved and
+will remove the need for special handling of sender_id = 0 on obs messages.
+
+### 2. sender_id = device_uuid & 0xFFFF
+
+We need to decide where sender_id will come from on Piksi Multi. In deciding
+this, we need to keep in mind the following goals:
+- sender_id *does not* need to be globally unique amongst all Piksi Multi's
+- sender_id *does* need to be unique amongst the Piksi Multi's on a customer's
+local network of devices (for instance, point to multi-point ISM radio network,
+TCP/IP network, etc)
+
+### 3. Remove request / response messages where possible
+
+The following request response messages are not used in Piksi Multi and can be
+deprecated:
+
+- `MSG_BOOTLOADER_HANDSHAKE_REQ`
+- `MSG_BOOTLOADER_HANDSHAKE_RESP`
+- `MSG_FLASH_READ_REQ`
+- `MSG_FLASH_READ_RESP`
+- `MSG_FLASH_PROGRAM`
+- `MSG_FLASH_ERASE`
+- `MSG_FLASH_DONE`
+- `MSG_STM_FLASH_LOCK_SECTOR`
+- `MSG_STM_FLASH_UNLOCK_SECTOR`
+- `MSG_M25_FLASH_WRITE_STATUS`
+
+The following messages could be part of a device ID message that is
```

Comment:
But we can't get rid of polling messags, because we still need to have some polling-based messages (like settings), so why not just keep these DNA, etc. messages similarly accessible?